tools/codegen/dest/register_dispatch_key.py

```diff
@@ -599,13 +607,30 @@ def generate_defn(cpp_sig: CppSignature) -> str:
                 # I didn't do it for this version
                 sig_body.append(f"at::{api_name}({out_exprs});")
             elif self.backend_index.dispatch_key != DispatchKey.Meta:
-                impl_exprs = ', '.join(
-                    e.expr for e in translate(
-                        context,
-                        structured.impl_arguments(self.g),
-                        method=False
-                    )
-                )
+                impl_exprs_gen = (e.expr for e in translate(context, structured.impl_arguments(self.g), method=False))
```

Comment:
no change necessary here, but fwiw, I'd probably personally just have this be a list rather than a generator, because the risk of accidentally using the generator multiple times is too big.

Reply:
Also this is predicated on `impl_arguments` actually returning the correct set of required arguments. So if you do the refactoring I suggested you actually have to do it this way, so it's not that big brain anymore